diff --git a/repository/src/main/scala/org/apache/atlas/query/QueryParser.scala b/repository/src/main/scala/org/apache/atlas/query/QueryParser.scala
index 233a3e8..42d76c8 100755
--- a/repository/src/main/scala/org/apache/atlas/query/QueryParser.scala
+++ b/repository/src/main/scala/org/apache/atlas/query/QueryParser.scala
@@ -73,6 +73,8 @@ trait QueryKeywords {
     protected val MAX = Keyword("max")
     protected val MIN = Keyword("min")
     protected val SUM = Keyword("sum")
+    protected val BY = Keyword("by")
+    protected val ORDER = Keyword("order")
 }
 
 trait ExpressionUtils {
@@ -270,7 +272,7 @@ object QueryParser extends StandardTokenParsers with QueryKeywords with Expressi
     def fromSrc = identifier ~ AS ~ alias ^^ { case s ~ a ~ al => s.as(al)} |
         identifier
 
-    def orderby = ORDERBY ~ expr ~ opt (asce) ^^ {
+    def orderby = (ORDERBY|(ORDER ~ BY )) ~ expr ~ opt (asce) ^^ {
       case o ~ odr ~ None => (odr, true)
       case o ~ odr ~ asc => (odr, asc.get)
     }
diff --git a/repository/src/test/scala/org/apache/atlas/query/ParserTest.scala b/repository/src/test/scala/org/apache/atlas/query/ParserTest.scala
index 8f277fc..878b0e3 100755
--- a/repository/src/test/scala/org/apache/atlas/query/ParserTest.scala
+++ b/repository/src/test/scala/org/apache/atlas/query/ParserTest.scala
@@ -92,4 +92,8 @@ class ParserTest extends BaseTest {
     )
   }
 
+  @Test def testorder_by: Unit = {
+    println(QueryParser.apply("from DB order by columnA").right.get.toString)
+  }
+
 }
